tests/ssl-opt.sh

```diff
@@ -683,8 +702,17 @@ requires_openssl_tls1_3() {
             OPENSSL_TLS1_3_AVAILABLE="NO"
         fi
     fi
+
     if [ "$OPENSSL_TLS1_3_AVAILABLE" = "NO" ]; then
         SKIP_NEXT="YES"
+    else
+        if [ "$OPENSSL_3_X_AVAILABLE" = "NO" ]; then
+            if is_config_enabled "MBEDTLS_ECDH_C"; then
+                SKIP_NEXT="NO"
```

Comment:
This is no correct, a `requires_xxx` function should never set `SKIP_NEXT` to `NO`, it should either set it to `YES` or leave it unchanged: the default value is `NO` but this may have been changed to `YES` by a previous call to another `requires_yyy` function, in which case it must remain unchanged.

Reply:
Note: this is what makes the CI fail in `component_test_default_cmake_gcc_asan` and all other components that don't have TLS 1.3 enabled: the various `TLS 1.3 O->m: HRR` test are supposed to be skipped due to
```
requires_config_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED
requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
```
which both the `SKIP_NEXT="YES"` but this function incorrectly sets it back to `NO`, so the test is not skipped as it should and fails instead.

Reply:
Thanks I was looking also into these failures and couldn't understand what is going on.